### PR TITLE
[MAINTENANCE] Update ruff to `0.0.269` and target `python 3.8`

### PR DESCRIPTION
**Pull request by Kilo59**
Updates the minimum version of python ruff targets from `3.7` -> `3.8`.

This causes many auto-fix rules to be triggered relating to `typing_extentions` imports now able to be imported using the standard lib `typing` module.

https://github.com/charliermarsh/ruff/releases/tag/v0.0.269

I will followup this PR with a `.git-blame-ignore-revs` entry

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -207,7 +207,7 @@ warn_required_dynamic_aliases = true
 warn_untyped_fields = true
 
 [tool.ruff]
-target-version = "py37"
+target-version = "py38"
```
Comment (Kilo59) [inline pyproject.toml:210]: Ruff now targets `3.8` instead of `3.7`

Comment (Shinnnyshinshin) [reply pyproject.toml:210]: 🚀

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ repos:
     hooks:
       - id: black-jupyter
   - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: 'v0.0.265'
```
Comment (Shinnnyshinshin) [inline .pre-commit-config.yaml:20]: 🚀

**Review by Shinnnyshinshin — APPROVED** [review]
Bring on the future 👍🏼

**Pull request merged**